app/controllers/articles_controller.rb

```diff
@@ -1,15 +1,30 @@
 class ArticlesController < ApplicationController
   before_action :authorized, only: [:new, :create]
+  before_action :set_article, only: [:show, :upvote, :downvote]
```

Comment:
Shouldn't have to be calling a before action to set article, just set it when you call it like I've shown you before.